src/components/dialog/Dialog.a11y.spec.jsx

```diff
@@ -95,4 +99,23 @@ describe('Dialog a11y', () => {
     userEvent.click(dialog.getByText(buttonText));
     waitForElementToBeRemoved(dialog.queryByRole('dialog'));
   });
+
+  it('correctly handles escape key for nested Dialogs', () => {
+    const onDismissOuter = jest.fn();
```

Comment:
Instead of testing calling function the better approach would be to test effect. The effect should be that inner dialog closes while outer dialog stays.

Reply:
Discussed on Slack, we've decided to stay with this version for the time being until better solution will be found